core/src/main/java/io/spine/core/Events.java

```diff
@@ -218,6 +219,52 @@ public static boolean isExternal(EventContext context) {
     }
 
     /**
+     * Obtains the compacted version of the event, which should be used for storing.
+     *
+     * <p>A compacted version doesn't contain:
+     * <ul>
+     *     <li>the enrichment from the event context;</li>
+     *     <li>the enrichment from the origin;</li>
+     *     <li>nested origins if the origin is {@link EventContext}.</li>
+     * </ul>
+     *
+     * @param event the event to compact
+     * @return the compacted event
+     */
+    @Internal
+    public static Event compact(Event event) {
+        final EventContext context = event.getContext();
+        final EventContext.Builder resultContext = context.toBuilder()
+                                                          .clearEnrichment();
+        final EventContext.OriginCase originCase = resultContext.getOriginCase();
+        switch (originCase) {
+            case EVENT_CONTEXT:
+                resultContext.setEventContext(context.getEventContext()
+                                                     .toBuilder()
+                                                     .clearOrigin()
+                                                     .clearEnrichment());
+                break;
+            case REJECTION_CONTEXT:
+                resultContext.setRejectionContext(context.getRejectionContext()
+                                                         .toBuilder()
+                                                         .clearEnrichment());
+                break;
+            case COMMAND_CONTEXT:
+                // Nothing to remove.
+                break;
+            case ORIGIN_NOT_SET:
+                // Does nothing because there is no origin for this event.
+                break;
+            default:
+                throw newIllegalStateException("Unsupported origin case is encountered: %s",
```

Comment:
As long as this method is `public` this branch can be tested. Please do.

Reply:
It cannot be tested even with using of reflection.